js/ui/scheduler/ui.scheduler.appointments.strategy.base.js

```diff
@@ -506,7 +510,96 @@ var BaseRenderingStrategy = Class.inherit({
         return COMPACT_APPOINTMENT_DEFAULT_OFFSET;
     },
 
-    getAppointmentDataCalculator: noop
+    getAppointmentDataCalculator: noop,
+
+    _customizeCoordinates: function(coordinates, ratio, appointmentCountPerCell, maxHeight, isAllDay) {
+        var index = coordinates.index,
+            height = ratio * maxHeight / appointmentCountPerCell,
+            top = (1 - ratio) * maxHeight + coordinates.top + (index * height),
+            width = coordinates.width,
+            left = coordinates.left,
+            compactAppointmentDefaultSize,
+            compactAppointmentDefaultOffset;
+
+        if(coordinates.isCompact) {
+            compactAppointmentDefaultSize = this.getCompactAppointmentDefaultSize();
+            compactAppointmentDefaultOffset = this.getCompactAppointmentDefaultOffset();
+            top = coordinates.top + compactAppointmentDefaultOffset;
+            left = coordinates.left + (index - appointmentCountPerCell) * (compactAppointmentDefaultSize + compactAppointmentDefaultOffset) + compactAppointmentDefaultOffset;
+            height = compactAppointmentDefaultSize;
+            width = compactAppointmentDefaultSize;
+
+            this._markAppointmentAsVirtual(coordinates, isAllDay);
+        }
+
+        return {
+            height: height,
+            width: width,
+            top: top,
+            left: left
+        };
+    },
+
+    _calculateGeometryConfig: function(coordinates) {
+        var overlappingMode = this.instance.fire("getMaxAppointmentsPerCell"),
+            offsets = this._getOffsets();
+
+        var appointmentCountPerCell = this._getAppointmentCount(overlappingMode, coordinates);
+        var ratio = this._getDefaultRatio(coordinates, appointmentCountPerCell);
+        var maxHeight = this._getMaxHeight();
+
+        if(!appointmentCountPerCell) {
+            appointmentCountPerCell = coordinates.count;
+            ratio = (maxHeight - offsets.unlimited) / maxHeight;
+        }
+        if(overlappingMode === "auto") {
+            ratio = (maxHeight - offsets.auto) / maxHeight;
+        }
+
+        return {
+            ratio: ratio,
+            appointmentCountPerCell: appointmentCountPerCell,
+            maxHeight: maxHeight
+        };
+    },
+
+    _getAppointmentCount: noop,
+
+    _getDefaultRatio: noop,
+
+    _getOffsets: noop,
+
+    _getMaxHeight: noop,
+
+    _needVerifyItemSize: function() {
+        return false;
+    },
+
+    _getAppointmentCountPerCell: function() {
```

Comment:
_getAppointmentCountPerCell method does not return an 'getMaxAppointmentsPerCell' option value, it does some calculations on this option. But maybe it makes sense to rename the method.